# This patch is taken from cups 1.4.5 and fixes annoying requirement to have poppler around during build-time

--- cups-1.3.11/config-scripts/cups-pdf.m4	2009-03-12 22:34:21.000000000 +0100
+++ cups-1.4.5/config-scripts/cups-pdf.m4	2009-08-08 00:30:30.000000000 +0200
@@ -13,12 +13,7 @@
 dnl   file is missing or damaged, see the license at "http://www.cups.org/".
 dnl
 
-AC_ARG_ENABLE(pdftops, [  --enable-pdftops        build pdftops filter, default=auto ])
-AC_ARG_WITH(pdftops, [  --with-pdftops          set pdftops filter (gs,pdftops,none), default=pdftops ])
-
-if test "x$enable_pdftops" = xno -a "x$with_pdftops" = x; then
-	with_pdftops=no
-fi
+AC_ARG_WITH(pdftops, [  --with-pdftops          set pdftops filter (gs,/path/to/gs,pdftops,/path/to/pdftops,none), default=pdftops ])
 
 PDFTOPS=""
 CUPS_PDFTOPS=""
@@ -52,6 +47,12 @@
 	fi
 	;;
 
+	x/*/gs) # Use /path/to/gs without any check:
+	CUPS_GHOSTSCRIPT="$with_pdftops"
+	AC_DEFINE(HAVE_GHOSTSCRIPT)
+	PDFTOPS="pdftops"
+	;;
+
 	xpdftops)
 	AC_PATH_PROG(CUPS_PDFTOPS, pdftops)
 	if test "x$CUPS_PDFTOPS" != x; then
@@ -62,6 +63,20 @@
 		exit 1
 	fi
 	;;
+
+	x/*/pdftops) # Use /path/to/pdftops without any check:
+	CUPS_PDFTOPS="$with_pdftops"
+	AC_DEFINE(HAVE_PDFTOPS)
+	PDFTOPS="pdftops"
+	;;
+
+	xnone) # Make no pdftops filter if with_pdftops=none:
+	;;
+
+	*) # Invalid with_pdftops value:
+	AC_MSG_ERROR(Invalid with_pdftops value!)
+	exit 1
+	;;
 esac
 
 AC_DEFINE_UNQUOTED(CUPS_PDFTOPS, "$CUPS_PDFTOPS")
